### ISSUE TITLE
Dealing with NA values in geojson_write, as.json etc.

### ISSUE THREAD

**Issue by ateucher** [ropensci/geojsonio#109]
When there are `NA` values in a field, `jsonlite::toJSON` deals with them differently depending on the type. I recently got bitten by this for numeric values where values that were `NA` in R were written to geojson file as `"NA"` rather than `null` as I would have expected.

``` r
library(geojsonio)
mylist <- list(list(latitude=30, longitude=120, foo=NA_real_),
               list(latitude=30, longitude=130, foo=5.3))
as.json(mylist)
#> [{"latitude":30,"longitude":120,"foo":"NA"},{"latitude":30,"longitude":130,"foo":5.3}]
```

I'm not sure when you *wouldn't* want `null` for `NA` values, but I'm happy to be educated 😄 

We could hardcode `na = "null"` in `jsonlite::toJSON` in `as.json`, or pass `...` all the way through from all exported functions that use `as.json` so the user can decide?  Could be confusing for `geojson_write`, as currently `...` is passed to `writeOGR` for those methods that use it.

**Comment by sckott** [comment]
Hmm, i agree that `null` makes more sense - and I'd vote to change it to `na = "null"` by default. could add new param of `na` that's passed on to `jsonlite::toJSON`?

**Comment by ateucher** [comment]
> could add new param of `na` that's passed on to `jsonlite::toJSON`?

Good idea. Also wonder if we should add a `pretty` param that is passed through... With big geojson files I often wish for that option.

**Comment by sckott** [comment]
pretty is passed to `jsonlite::toJSON` ?

**Comment by ateucher** [comment]
Yep. It's hardcoded to `TRUE` right now in `geo_list` and `json` methods of `geojson_write`: https://github.com/ropensci/geojsonio/blob/master/R/geojson_write.r#L390-#L409

**Comment by ateucher** [comment]
But `geojson_json` `...` is passed through to `jsonlite::toJSON` so you could use `pretty` there - that is a bit different.

**Comment by sckott** [comment]
okay

> Also wonder if we should add a pretty param that is passed through... With big geojson files I often wish for that option.

so do you mean with big files you'd prefer to have pretty as `TRUE` or `FALSE`, just curious, i assume `FALSE` to save disk space due to newlines and such removed?

**Comment by ateucher** [comment]
Yup, you're right. It'd be nice to be able to minify as much as possible with `pretty = FALSE`

**Comment by sckott** [comment]
okay

**Comment by ateucher** [comment]
Any thoughts on how best to document `geojson_write` and `geojson_json` where some methods (e.g., sf, json, geo_list) use `jsonlite::toJSON` internally and some (Spatial etc) use `rgdal::writeOGR` internally? `na` and `pretty` will have no effect on the latter.

**Comment by ateucher** [comment]
The other option is to just use `...` so a user can set `na`, `pretty` or whatever else they want - then in the docs say which methods pass `...` to `writeOGR`, and which pass it to `toJSON`. The docs already say `...` is passed to `writeOGR` which is only partially true. Unfortunately I don't think there's an elegant way to set a default (i.e., `na = "null"`) that can be overridden by an arg in `...`

**Comment by sckott** [comment]
do you lean one way or the other?  I think i lean towards `...` and documenting in which cases it goes to jsonlite and which to writeOGR

**Comment by ateucher** [comment]
I have the same inclination

**Comment by sckott** [comment]
@ateucher is this sorted out now, that is. do we close?

**Comment by ateucher** [comment]
Yep. Closed by #111
